X10D/src/Core/IntrinsicUtility.cs

```diff
@@ -0,0 +1,222 @@
+            // https://stackoverflow.com/questions/17863411/sse-multiplication-of-2-64-bit-integers
+
+            Vector256<ulong> ac = Avx2.Multiply(lhs.AsUInt32(), rhs.AsUInt32());
+            Vector256<uint> b = Avx2.ShiftRightLogical(lhs, 32).AsUInt32();
+            Vector256<ulong> bc = Avx2.Multiply(b, rhs.AsUInt32());
+            Vector256<uint> d = Avx2.ShiftRightLogical(rhs, 32).AsUInt32();
+            Vector256<ulong> ad = Avx2.Multiply(lhs.AsUInt32(), d);
+            Vector256<ulong> high = Avx2.Add(bc, ad);
+            high = Avx2.ShiftLeftLogical(high, 32);
+
+            return Avx2.Add(high, ac);
+        }
+
+        throw new PlatformNotSupportedException("Unsupported SIMD platform.");
+    }
+
+    /// <summary>
+    ///     <para>Multiply packed 64-bit signed integer elements in a and b and truncate the results to 64-bit integer.</para>
+    ///     <para>Operation:</para>
+    ///     <code>
+    ///     dest[0] = lhs[0] * rhs[0];
+    ///     dest[1] = lhs[1] * rhs[1];
+    ///     </code>
+    /// </summary>
+    /// <param name="lhs">Left vector.</param>
+    /// <param name="rhs">Right vector.</param>
+    /// <returns></returns>
+    /// <remarks>API avaliable on SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, AVX, AVX2, ARM NEON (untested) hardwares.</remarks>
+    [Pure]
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static Vector128<long> Multiply(Vector128<long> lhs, Vector128<long> rhs)
+    {
+        return Multiply(lhs.AsUInt64(), rhs.AsUInt64()).AsInt64();
+    }
+
+    /// <summary>
+    ///     <br>Multiply packed 64-bit signed integer elements in a and b and truncate the results to 64-bit integer.</br>
+    ///     <br>Operation:</br>
+    ///     <code>
+    ///     dest[0] = lhs[0] * rhs[0];
+    ///     dest[1] = lhs[1] * rhs[1];
+    ///     dest[2] = lhs[2] * rhs[2];
+    ///     dest[3] = lhs[3] * rhs[3];
+    ///     </code>
+    /// </summary>
+    /// <param name="lhs">Left vector.</param>
+    /// <param name="rhs">Right vector.</param>
+    /// <returns></returns>
+    /// <remarks>API avaliable on AVX2 hardware.</remarks>
+    [Pure]
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static Vector256<long> Multiply(Vector256<long> lhs, Vector256<long> rhs)
+    {
+        return Multiply(lhs.AsUInt64(), rhs.AsUInt64()).AsInt64();
+    }
+
+    /// <summary>
+    ///     <para>Horizontally apply OR operation on adjacent pairs of single-precision (32-bit) floating-point elements in lhs and rhs.</para>
+    ///     <para>Operation:</para>
+    ///     <code>
+    ///     dest[0] = lhs[0] | lhs[1];
+    ///     dest[1] = lhs[2] | lhs[3];
+    ///     dest[2] = rhs[0] | rhs[1];
+    ///     dest[3] = rhs[2] | rhs[3];
+    ///     </code>
+    /// </summary>
+    /// <param name="lhs">Left vector.</param>
+    /// <param name="rhs">Right vector.</param>
+    /// <returns></returns>
```

Comment:
You get the gist kek. The `returns` element on the xmldoc is empty. Also `PlatformNotSupportedException` is not documented